app/presenters/product_mailchimp_presenter.rb

```diff
@@ -14,7 +14,7 @@ def json
           id: Digest::MD5.hexdigest(product.id.to_s),
           title: product.name || "",
           description: product.description || "",
-          url: "#{ENV['BASE_URL']}/#{product.category&.permalink || 'products'}/#{product.slug}",
+          url: "#{Rails.application.routes.url_helpers.spree_url}/#{product.category&.permalink || 'products'}/#{product.slug}" || "",
```

Comment:
@chhga let's use `spree_product_url` helper here

Reply:
@damianlegawiec there is no `spree_product_url` method for `Rails.application.routes.url_helpers`. And we cannot use `Spree::Core::Engine.routes.url_helpers`